core/commonMain/src/kotlinx/serialization/internal/PluginUtils.kt

```diff
@@ -0,0 +1,45 @@
+package kotlinx.serialization.internal
+
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.InternalSerializationApi
+import kotlinx.serialization.MissingFieldException
+import kotlinx.serialization.descriptors.SerialDescriptor
+
+@OptIn(ExperimentalSerializationApi::class)
+@InternalSerializationApi
+public fun throwMissingFieldException(seen: Int, goldenMask: Int, descriptor: SerialDescriptor) {
+    val missingFields = mutableListOf<String>()
+    var seenBits = seen
+    var goldenMaskBits = goldenMask
+    for (i in 0 until 32) {
+        if ((goldenMaskBits and 1 != 0) && (seenBits and 1 == 0)) {
```

Comment:
It's also possible to get all reauired-and-missing bits set via `!(!goldenMaskBits or seenBits)` to replace two conditions in `if` with one. Not sure if this microoptimization has real benefit though.

Reply:
ok, thanks

Reply:
So it can be simplified to (goldenMaskBits and !seenBits)